repos/system_upgrade/el8toel9/actors/firewalldcheckservicetftpclient/actor.py

```diff
@@ -0,0 +1,35 @@
+from leapp import reporting
+from leapp.actors import Actor
+from leapp.models import FirewalldUsedObjectNames
+from leapp.reporting import create_report, Report
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+
+
+class FirewalldCheckServiceTftpClient(Actor):
+    '''
+    No documentation has been provided for the firewalld_check_service_tftp_client actor.
+    '''
+
+    name = 'firewalld_check_service_tftp_client'
+    consumes = (FirewalldUsedObjectNames,)
+    produces = (Report,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        tftp_client_service = 'tftp-client'
+        send_report = False
+
+        for facts in self.consume(FirewalldUsedObjectNames):
+            if tftp_client_service in facts.services:
+                send_report = True
+
+        if send_report:
+            create_report([
+                reporting.Title('Unsupported Firewalld Configuration'),
+                reporting.Summary('Firewalld has enabled service '
+                                  '"{service}" which has been removed in RHEL-9.'.format(
+                                      service=tftp_client_service)),
+                reporting.Severity(reporting.Severity.HIGH),
+                reporting.Tags([reporting.Tags.SANITY, reporting.Tags.FIREWALL]),
+                reporting.Flags([reporting.Flags.INHIBITOR])
+            ])
```

Comment:
@erig0 Couple of tips when creating report based on our experience with customers, so they will not be opening bunch of customer tickets & possibly bugs for that. Not saying you have to change it, just giving you tips how you could improve it if you want to improve UX.

it's good to ask yourself generic questions like:
* what is user supposed to do after the reading the report?
* what I want to tell to the user by this report?
* is it something we have documented / we are going to document in the official RH docs so there is URL with more details? (`reporting.ExternalLink`)

In my case it seems obvious that in case I want to continue with the upgrade, I have to disable the service. But from our experience, some users will not have an idea what they are supposed to do. Adding the remediation instructions can reduce amount of opened support tickets. Remediation instructions:
* with the hint field, like: in case you want to proceed the upgrade and you are ok the service is missing on the target system, stop & disable the service before the upgrade
* it's possible to provide the command field as well, which can be used directly via satellite or cockpit, but it has to be written now in the way it's just ONE command; e.g. in case of stop & disable the service, something like `bash -c "cmd1 && cmd2;"` because of the way how the current satellite/foreman plugin is written.

For inspiration abour remediations instructions see e.g. https://github.com/oamg/leapp-repository/blob/master/repos/system_upgrade/el7toel8/actors/checkhacluster/libraries/checkhacluster.py

In this case you can also add `reporting.RelatedResource('service', tftp_client_service)`

Reply:
+1, it would be great to give user some pointer how to proceed. Ideally a remediation command that could be applied, but at least description\general guidance.

Reply:
I added a Remediation, but there is no generic command to resolve the situation. The commands to resolve depends on the users config.